velox/functions/prestosql/Slice.cpp

```diff
@@ -202,14 +202,17 @@ class SliceFunction : public exec::VectorFunction {
 };
 
 static std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
-  return {// array(T, bigint, bigint) -> array(T)
-          exec::FunctionSignatureBuilder()
-              .typeVariable("T")
-              .returnType("array(T)")
-              .argumentType("array(T)")
-              .argumentType("bigint")
-              .argumentType("bigint")
-              .build()};
+  std::vector<std::shared_ptr<exec::FunctionSignature>> signatures;
+  for (auto type : {"integer", "bigint"}) {
```

Comment:
Hi @boneanxs, could you help check if integer is supported for Presto slice function? If not, it would be better to register  a slice function under sparksql with integer type instead.

Reply:
We can register a slice function reusing the Presto implementation with all the needed types of Spark.

Reply:
done